src/Hashids.net/Prev/Hashids.cs

```diff
@@ -0,0 +1,471 @@
+
+                stringBuilder.Insert(0, guard);
+
+                if (stringBuilder.Length < _minHashLength)
+                {
+                    guardIndex = (numbersHashInt + stringBuilder[2]) % _guards.Length;
+                    guard = _guards[guardIndex];
+
+                    stringBuilder.Append(guard);
+                }
+            }
+
+            var halfLength = _alphabet.Length / 2;
+
+            while (stringBuilder.Length < _minHashLength)
+            {
+                alphabet.CopyTo(shuffleBuffer);
+                ConsistentShuffle(alphabet, shuffleBuffer);
+
+#if NETSTANDARD2_0
+                stringBuilder.Insert(0, alphabet.Slice(halfLength, _alphabet.Length - halfLength).ToArray());
+                stringBuilder.Append(alphabet.Slice(0, halfLength).ToArray());
+#else
+                stringBuilder.Insert(0, alphabet[halfLength.._alphabet.Length]);
+                stringBuilder.Append(alphabet[..halfLength]);
+#endif
+
+                var excess = stringBuilder.Length - _minHashLength;
+                if (excess > 0)
+                {
+                    stringBuilder.Remove(0, excess / 2);
+                    stringBuilder.Remove(_minHashLength, stringBuilder.Length - _minHashLength);
+                }
+            }
+
+            var result = stringBuilder.ToString();
+            StringBuilderPool.Return(stringBuilder);
+            return result;
+        }
+
+        private int BuildReversedHash(long input, ReadOnlySpan<char> alphabet, Span<char> hashBuffer)
+        {
+            var length = 0;
+            do
+            {
+                int idx = (int)(input % _alphabet.Length);
+                hashBuffer[length] = alphabet[idx];
+                length += 1;
+                input /= _alphabet.Length;
+            }
+            while (input > 0);
+
+            return length;
+        }
+
+        private long Unhash(string input, ReadOnlySpan<char> alphabet)
+        {
+            long number = 0;
+
+            for (var i = 0; i < input.Length; i++)
+            {
+                var pos = alphabet.IndexOf(input[i]);
+                number = (number * _alphabet.Length) + pos;
+            }
+
+            return number;
+        }
+
+        private long[] GetNumbersFrom(string hash)
+        {
+            if (string.IsNullOrWhiteSpace(hash))
+                return Array.Empty<long>();
+
+            var hashArray = hash.Split(_guards, StringSplitOptions.RemoveEmptyEntries);
+            if (hashArray.Length == 0)
+                return Array.Empty<long>();
+
+            var unguardedIdx = (hashArray.Length is 3 or 2) ? 1 : 0;
+            var hashBreakdown = hashArray[unguardedIdx];
+
+            var lottery = hashBreakdown[0];
+            if (lottery == '\0') // default(char) is '\0'
+                return Array.Empty<long>();
+
+            hashArray = hashBreakdown.Substring(1).Split(_seps, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Ah you could get rid of the splits and substring.

Your welcome to use the implementation I did in #65 to avoid memory allocation.
Please include `Co-authored-by: Joseph Petersen <josephp90@gmail.com>`

https://github.com/ullmark/hashids.net/pull/65/files#diff-3d353b5b364e3fa95824e03e78bbd82a7f71a075d83e1942139b921ef40d40f1R651